src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [yearKey, monthKey, dayKey, separator] = fromFormat;
+  const splittedDate = date.split(separator);
+
+  const oldDateObj = [yearKey, monthKey, dayKey].reduce((acc, key, index) => {
+    acc[key] = splittedDate[index];
+
+    return acc;
+  }, {});
```

Comment:
The `reduce` function here is used to map the date parts to their respective keys. However, it's not necessary to use `reduce` for this purpose, as it adds complexity. You could achieve the same result with a simpler approach by directly assigning the values to the `oldDateObj` properties.